### ISSUE TITLE
False positive NRT warning in `when` clause

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#46070]
```cs
#nullable enable
public class C {
    public void M(string? s1, string? s2) {
        switch ((s1, s2))
        {
            case (null, _):
            case (_, _) when s1.Equals(s2): // warning: s1 could be null
                break;
        }
    }
}
```

`s1` being null is fully covered by the previous pattern and can't be null in that when clause.

/cc @gafter

**Comment by 333fred** [comment]
Actually nevermind, I completely forgot that NRT doesn't carry though a tuple literal like this.
